lisp/ess-utils.el

```diff
@@ -584,49 +590,35 @@ See also `ess-use-ido'."
       ;; else usual completion
       (completing-read prompt collection predicate require-match initial-input hist def))))
 
-(defun ess-load-extras (&optional inferior)
-  "Load all the extra features depending on custom settings."
-
-  (let ((mode (if inferior 'inferior-ess-mode 'ess-mode))
-        (isR (string-match "^R" ess-dialect)))
-
-    ;; auto-complete
-    (when (and (boundp 'ac-sources)
-               (if inferior
-                   (eq ess-use-auto-complete t)
-                 ess-use-auto-complete))
-      (add-to-list 'ac-modes mode)
-      ;; files should be in front; ugly, but needed
-      (when ess-ac-sources
-        (setq ac-sources
-              (delq 'ac-source-filename ac-sources))
-        (mapc (lambda (el) (add-to-list 'ac-sources el))
-              ess-ac-sources)
-        (add-to-list 'ac-sources 'ac-source-filename)))
-
-    ;; company
-    (when (and (boundp 'company-backends)
-               (if inferior
-                   (eq ess-use-company t)
-                 ess-use-company))
-      (when ess-company-backends
-        (set (make-local-variable 'company-backends)
-             (cl-copy-list (append ess-company-backends company-backends)))
-        (delq 'company-capf company-backends)))
-
-    ;; eldoc)
-    (require 'eldoc)
-    (when (and ess-eldoc-function ;; if mode provide this, it suports eldoc
-               (or (and (not inferior) ess-use-eldoc)
-                   (and inferior (eq ess-use-eldoc t))))
-      (when (> eldoc-idle-delay 0.4) ;; default is too slow for paren help
-        (set (make-local-variable 'eldoc-idle-delay) 0.1))
-      (set (make-local-variable 'eldoc-documentation-function) ess-eldoc-function)
-      (eldoc-mode 1))
-
-    ;; tracebug
-    (when (and ess-use-tracebug inferior isR (fboundp 'ess-tracebug))
-      (ess-tracebug 1))))
+(defun ess--maybe-setup-auto-complete (sources &optional inferior)
```

Comment:
Can we please get rid of "maybe"? "Setup" can mean "do nothing" if the settings is so.

Reply:
Sure, I'll go back and change it. I wrote this before I knew of your hatred of that word :-)